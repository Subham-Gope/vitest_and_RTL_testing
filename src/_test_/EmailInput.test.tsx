// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import EmailInput from "../components/EmailInput";

// describe("Testing the EmailInput.tsx file", () => {
//   test("simulated typing  in the input", async () => {
//     const { container } = render(
//       <EmailInput
//         message={`This is testing the userEvent.type feature`}
//         assignment={1}
//       />
//     );

//     expect(container).toMatchSnapshot();

//     const emailInput = screen.getByLabelText("Enter Email");

//     expect(emailInput).toBeInTheDocument();

//     await userEvent.type(emailInput, "user@example.com");

//     expect(emailInput).toHaveValue("user@example.com");
//   });
// });
