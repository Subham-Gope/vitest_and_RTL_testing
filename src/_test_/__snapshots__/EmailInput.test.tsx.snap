// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the EmailInput.tsx file > simulated typing  in the input 1`] = `
<div>
  <p>
    This is testing the userEvent.type feature
     - Testing assignment number: 
    1
  </p>
  <div>
    <label
      for="email-input"
    >
      Enter Email 
          
      <input
        aria-label="Email-Input"
        id="email-input"
        placeholder="xyz@email.com"
        type="email"
        value=""
      />
    </label>
  </div>
</div>
`;
